### ISSUE TITLE
IWR1443 on MAC OS X

### ISSUE THREAD

**Issue by nexusz** [ibaiGorordo/IWR1443-Read-Data-Python#5]
Hi,

I tried your reader on Mac OS X. I only needed to change the serial ports
    # MAC
    CLIport = serial.Serial('/dev/tty.usbmodemR0031201', 115200)
    Dataport = serial.Serial('/dev/tty.usbmodemR0031204', 921600)

That is the only change I needed to make.

The sdk firmware I uploaded to the board is version mmwave_sdk_01_02_00_05.

I started the reader. The graphical screen appears, and I can see data being plotted. However, after a few seconds (sometimes a bit longer, but no more than 10 seconds) the reader crashes and the following exception is given:

  File "readData_IWR1443.py", line 334, in <module>
    dataOk = update()
  File "readData_IWR1443.py", line 292, in update
    dataOk, frameNumber, detObj = readAndParseData14xx(Dataport, configParameters)
  File "readData_IWR1443.py", line 147, in readAndParseData14xx
    byteBuffer[:byteBufferLength-startIdx[0]] = byteBuffer[startIdx[0]:byteBufferLength]
ValueError: could not broadcast input array from shape (0) into shape (32700)

I'm not a python expert at all, but to me this reads as an undesirable condition is found and therefore the exception occurs.
Can you tell me how to cope with it?

Thanks in advance

Below the complete output:
=======

% ***************************************************************
% Created for SDK ver:01.02
% Created using Visualizer ver:2.0.0.2
% Frequency:77
% Platform:xWR14xx
% Scene Classifier:best_range_res
% Azimuth Resolution(deg):15 + Elevation
% Range Resolution(m):0.047
% Maximum unambiguous Range(m):2.41
% Maximum Radial Velocity(m/s):1
% Radial velocity resolution(m/s):0.13
% Frame Duration(msec):50
% Range Detection Threshold (dB):15
% Range Peak Grouping:disabled
% Doppler Peak Grouping:disabled
% Static clutter removal:enabled
% ***************************************************************
sensorStop
flushCfg
dfeDataOutputMode 1
channelCfg 15 7 0
adcCfg 2 1
adcbufCfg 0 1 0 1
profileCfg 0 77 284 7 40 0 0 100 1 64 2000 0 0 30
chirpCfg 0 0 0 0 0 0 0 1
chirpCfg 1 1 0 0 0 0 0 4
chirpCfg 2 2 0 0 0 0 0 2
frameCfg 0 2 16 0 50 1 0
lowPower 0 0
guiMonitor 1 0 0 0 0 0
cfarCfg 0 2 8 4 3 0 1195
peakGrouping 1 0 0 1 56
multiObjBeamForming 1 0.5
clutterRemoval 1
calibDcRangeSig 0 -5 8 256
compRangeBiasAndRxChanPhase 0.0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0 1 0
measureRangeBiasAndRxChanPhase 0 1.5 0.2
CQRxSatMonitor 0 3 4 99 0
CQSigImgMonitor 0 31 4
analogMonitor 1 1
sensorStart
Traceback (most recent call last):
  File "readData_IWR1443.py", line 334, in <module>
    dataOk = update()
  File "readData_IWR1443.py", line 292, in update
    dataOk, frameNumber, detObj = readAndParseData14xx(Dataport, configParameters)
  File "readData_IWR1443.py", line 147, in readAndParseData14xx
    byteBuffer[:byteBufferLength-startIdx[0]] = byteBuffer[startIdx[0]:byteBufferLength]
ValueError: could not broadcast input array from shape (0) into shape (32700)

**Comment by ibaiGorordo** [comment]
Hi,

I am glad to hear it also works in MAC OS. I am not sure what might be the problem (I am not a python expert neither), just in case I have edited the line 142 as it was suggested in #3 
`  if startIdx[0] > 0 and startIdx[0] < byteBufferLength:`

If it does not fix your problem, please print the values of **startIdx[0]** and of **byteBufferLength** to see what might be wrong.

Also, it seems your configuration is for 20 Hz, so try to change the sleep time to 0.05 in line 337. It might help.

Best regards,
Ibai

**Comment by nexusz** [comment]
Hi Ibai,

I tried your suggestions. They did not seem to work. The sleeptime of 0.05 results in a delay of input to output. Did not use it anymore,

I reversed to the original, and the problem that I had (crash of the reader) does not appear anymore, or I did not see it crash within 5 - 10 min.

I do have a resulting question:
the config you created is very much oriented at detecting close distance (30-50 cm) movement.
The image onscreen does not show blue dots when there is no movement, and does show blue dots when there is movement.

The configuration in the mmWave SDK is much more oriented towards detecting an objects from 0 - 10 meters.

My question: can you give me a pointer to how you setup the config file that you created?

**Comment by ibaiGorordo** [comment]
Hi,

I am glad to hear you don't have problems any more. I have no idea what might have been the problem.

In my application I am detecting objects within 1 meter, so that is why I made that configuration. But basically I just tried adjusting different values on the mmwave demo visualizer and observed what configuration gave the best results. 

It is not a very clear answer, but it was mostly trial and error.
